content/support/enablement/k8-migration.md

```diff
@@ -0,0 +1,101 @@
+>
+> - Repositories will be re-cloned
+> - Search indexes will be rebuilt from scratch
+>
+> The above may take a while if you have a lot of repositories. In the meantime, searches may be slow or return incomplete results. This process rarely takes longer than 6 hours and is usually **much** faster.
+
+> NOTE: In some places you will see `$NAMESPACE` used. Add `-n $NAMESPACE` to commands if you are not using the default namespace
+> More kubectl configuration options can be found here: [kubectl Cheat Sheet](https://kubernetes.io/docs/reference/kubectl/cheatsheet/)
+
+## Overview
+
+1. Backup Sourcegraph databases in the old cluster by generating dumps
+2. Restore the Dumps in the freshly deployed instance on another cluster.
+
+### Steps to backup Sourcegraph Databases
+
+In this section of the tutorial, you would learn how to back up the primary `sourcegraph` database and the `codeintel` database.
+
+- Verify that the deployment is running.
+
+  `kubectl get pods -A`
+
+- Stop all connections to the database by removing the frontend deployment.
+
+  `kubectl scale --replicas=0 deployment/sourcegraph-frontend`
+
+- Generate the database dumps.
+
+  - Dumps for the primary **sourcegraph** DB
+    `kubectl exec -it $pgsql_POD_NAME -- bash -c 'pg_dump -C --username sg sg' > sourcegraph_db.out`
+
+  - Dumps for the **codeintel** DB
+    `kubectl exec -it $codeintel-db_POD_NAME -- bash -c 'pg_dump -C --username sg sg' > codeintel_db.out`
+
+- Ensure you copy/move the `sourcegraph_db.out` and `codeintel_db.out` files to the root of the `deploy-sourcegraph` directory of the new Sourcegraph deployment.
+
+### Steps to restore Sourcegraph Databases into a new environment
+
+In this section of the tutorial, you would learn how to restore the primary `sourcegraph` database and the `codeintel` database from the generated dumps in the previous section.
+
+- Before restoring on the new deployment, It is very important you stop all connections to the Database from the frontend.
+
+  -     Scale down all the pods in the deployment
+    `kubectl scale deployment ---all --reliplicas=0 `
+  -     Scale down all the statefulset
+    `kubectl scale sts ---all --reliplicas=0 `
+
+- Then restart ONLY the `pgsql` and `codeintel` pods
+
+  -     Start the pgsql:
+    `kubectl scale pgsql --reliplicas=1 `
+  -     Start the codeintel-db:
+    `kubectl scale codeintel-db --reliplicas=1 `
+
+- Copy the database files into the pods by running the following command from the root of the deploy-sourcegraph directory
+
+  ` kubectl cp sourcegraph_db.out $pgsql_POD_NAME:/tmp/sourcegraph_db.out`
+
+  `kubectl cp codeintel_db.out $codeintel-db_POD_NAME:/tmp/codeintel_db.out`
+
+- Exec into the pgsql pod and then the template1 database:
+
+  ```
+  kubectl exec -it <pgsql-pod-name> -- sh
+  $ psql -U sg template1
+  ```
+
+- Drop the sg database, recreate it, then exit the psql session (but still in the pod exec):
+
+  ```
+  # DROP DATABASE sg;
+  # CREATE DATABASE sg;
+  # \q
```

Comment:
Just wanted to call out that this is something that we have to do to work around how Sourcegraph sets up the databases today. 
Our database pods automatically create the `sg` database on startup if it isn't present. Our liveness probes also check for it which means it not being present can cause `k8s` to kill the pod.